### PR TITLE
Adds six new strange object powers!

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

DESIGN CONCEPT: Strange Devices are at their best when they're odd tools which can help a department out, with some strange drawbacks that make them interesting and unique. Foam for the Janitor, Flashbang for Security, Teleporter for tiders, Corgimaker for the chef. They're also ambiguous enough that they can really be used by anyone for silly hijinks.

Drink Dispenser: Creates a glass, then fills it up with a random drink.

Bioscrambler: After a noticeable warning, bioscrambles people nearby. 33% chance to be used up on use.

Recharger: Shocks you, but recharges one random item to max battery.

Hugger: Hugs everyone in range as if you had the friendly trait.

DImensional: Converts the surroundings to a random material-theme. Each use increases cooldown by two seconds.

Disguiser: Drops your clothing to the ground and replaces it with a delete-on-drop costume and a randomly assigned cardboard ID. 

Removed an unused and unusable shock flag.

Removed an unused dumb ghost item.

Added a proc that returns a hex color with the # attached.

## Why It's Good For The Game

> DESIGN CONCEPT: Strange Devices are at their best when they're odd tools which can help a department out, with some strange drawbacks that make them interesting and unique. Foam for the Janitor, Flashbang for Security, Teleporter for tiders, Corgimaker for the chef. They're also ambiguous enough that they can really be used by anyone for silly hijinks.

Request by ArcaneMusic! He wanted new abilities to go along with the new sprites, and approved these.

> Removed an unused dumb ghost item.

please stop using costume subtypes for OP items. whats the damn point of the subtype if youre just going to screw with it

> Added a proc that returns a hex color with the # attached.

idk why it didnt have this tbh

## Changelog

:cl:
add: Adds six new strange object powers! Drink dispenser, bioscrambler, recharger, hugger, dimensional, disguiser.
code: Removed some unused code and items, and added a proc that returns a random #colorstring.
/:cl:

### CODE REVIEW

File: code/modules/research/experimentor.dm [hunk 1]
```diff
@@ -722,6 +728,165 @@
...
 
+// Creates a glass and fills it up with a drink.
+/obj/item/relic/proc/drink_dispenser(mob/user)
+	var/obj/item/reagent_containers/cup/glass/drinkingglass/freebie = new(get_step_rand(user))
+	playsound(get_turf(freebie), 'sound/effects/phasein.ogg', rand(25,50), TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	addtimer(CALLBACK(src, PROC_REF(dispense_drink), freebie), 0.5 SECONDS)
+
+/obj/item/relic/proc/dispense_drink(obj/item/reagent_containers/cup/glass/glasser)
+	glasser.reagents.add_reagent(get_random_drink_id(), rand(glasser.volume * 0.3, glasser.volume))
+	playsound(get_turf(glasser), SFX_SPARKS, rand(25,50), TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
```
Comment (MrMelbert) [inline code/modules/research/experimentor.dm:739]:
```suggestion
	playsound(glasser, SFX_SPARKS, rand(25,50), TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
```

File: code/modules/research/experimentor.dm [hunk 2]
```diff
@@ -722,6 +728,165 @@
 		throw_smoke(get_turf(user))
 		warn_admins(user, "Teleport", 0)
 
+// Creates a glass and fills it up with a drink.
+/obj/item/relic/proc/drink_dispenser(mob/user)
+	var/obj/item/reagent_containers/cup/glass/drinkingglass/freebie = new(get_step_rand(user))
+	playsound(get_turf(freebie), 'sound/effects/phasein.ogg', rand(25,50), TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
```
Comment (MrMelbert) [inline code/modules/research/experimentor.dm:734]:
```suggestion
	playsound(freebie, 'sound/effects/phasein.ogg', rand(25,50), TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
```

File: code/modules/research/experimentor.dm [hunk 3]
```diff
@@ -722,6 +728,165 @@
...
+// Charges an item or two in your inventory. Also yourself.
+/obj/item/relic/proc/charger(mob/living/user)
+	to_chat(user, span_danger("You're recharged!"))
+	var/stunner = 1.25 SECONDS
+	if(iscarbon(user))
+		var/mob/living/carbon/carboner = user
+		carboner.electrocute_act(15, src, flags = SHOCK_NOGLOVES, stun_duration = stunner)
+	else
+		user.electrocute_act(15, src, flags = SHOCK_NOGLOVES)
+	playsound(get_turf(user), SFX_SPARKS, rand(25,50), TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
```
Comment (MrMelbert) [inline code/modules/research/experimentor.dm:768]:
```suggestion
	playsound(user, SFX_SPARKS, rand(25,50), TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
```

File: code/modules/research/experimentor.dm [hunk 4]
```diff
@@ -722,6 +728,165 @@
...
+		new /obj/effect/temp_visual/heart(victim.loc)
+	if(length(huggeds))
+		to_chat(user, span_nicegreen("You feel friendly!"))
+	else
+		to_chat(user, pick(span_notice("You hug yourself, for some reason."), span_notice("You have a strange feeling for a moment, but then it passes.")))
+
+// Converts a 3x3 area into a random dimensional theme.
+/obj/item/relic/proc/dimensional_shift(mob/user)
+	var/datum/dimension_theme/shifter = pick(subtypesof(/datum/dimension_theme))
+	shifter = new shifter(src)
```
Comment (MrMelbert) [inline code/modules/research/experimentor.dm:814]: Pretty sure themes are singletons, you don't make new ones

File: code/modules/research/experimentor.dm [hunk 5]
```diff
@@ -722,6 +728,165 @@
...
+	carbonius.dropItemToGround(carbonius.head)
+	carbonius.equip_to_slot_or_del(disguise_hat, ITEM_SLOT_HEAD)
+	if(!ishuman(carbonius))
+		to_chat(user, span_notice("You have a peculiar feeling for a moment, but then it passes."))
+		return
+
+	var/mob/living/carbon/human/humerus = carbonius
+	// uniform
+	var/obj/item/clothing/under/costume/disguise_uniform = roll_costume(/obj/item/clothing/under/costume)
+	humerus.dropItemToGround(humerus.w_uniform)
```
Comment (MrMelbert) [inline code/modules/research/experimentor.dm:849]: Note `dropItemToGround` can fail, for nodrop items. Don't know if you care or not

File: code/modules/research/experimentor.dm [hunk 6]
```diff
@@ -722,6 +728,165 @@
...
+	card_id.scribbled_assignment = random_trim.assignment
+	card_id.update_overlays()
+	REMOVE_TRAIT(user, TRAIT_NO_JUMPSUIT, REF(src))
+
+/obj/item/relic/proc/roll_costume(type, flagcheck)
+	var/list/candidates = list()
+	for(var/obj/item/thingy as anything in subtypesof(type))
+		if(flagcheck && \
+		!(initial(thingy.flags_inv) & flagcheck))
+			continue
```
Comment (MrMelbert) [inline code/modules/research/experimentor.dm:881]: What is this multiline

File: code/modules/research/experimentor.dm [hunk 7]
```diff
@@ -722,6 +728,165 @@
...
+	new /obj/effect/temp_visual/bioscrambler_wave/light(get_turf(src))
+	to_chat(user, span_notice("Your stomach starts growling..."))
+	addtimer(CALLBACK(src, PROC_REF(scrambliticus), user), rand(1 SECONDS, 3 SECONDS)) // throw it away!
+
+/obj/item/relic/proc/scrambliticus(mob/user)
+	new /obj/effect/temp_visual/bioscrambler_wave(get_turf(src))
+	playsound(src, 'sound/magic/cosmic_energy.ogg', vol = 50, vary = TRUE)
+	for(var/mob/living/carbon/nearby in hearers(2, src))
+		nearby.bioscramble(name)
+		playsound(get_turf(nearby), SFX_SPARKS, rand(25,50), TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
```
Comment (MrMelbert) [inline code/modules/research/experimentor.dm:753]:
```suggestion
		playsound(nearby, SFX_SPARKS, rand(25,50), TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
```

File: code/__DEFINES/mobs.dm [hunk 8]
```diff
@@ -383,8 +383,6 @@
 #define SHOCK_SUPPRESS_MESSAGE (1 << 4)
 /// No skeleton animation if a human was shocked
 #define SHOCK_NO_HUMAN_ANIM (1 << 5)
-/// Ignores TRAIT_STUNIMMUNE
-#define SHOCK_IGNORE_IMMUNITY (1 << 6)
```
Comment (MrMelbert) [inline code/__DEFINES/mobs.dm:387]: Why remove it instead of making it not-unused as intended?

File: code/modules/clothing/suits/ghostsheet.dm [hunk 9]
```diff
@@ -21,9 +21,3 @@
 	. = ..()
 	if(!isinhands && check_holidays(HALLOWEEN))
 		. += emissive_appearance('icons/mob/simple/mob.dmi', "ghost", offset_spokesman = src, alpha = src.alpha)
-
-/obj/item/clothing/suit/costume/ghost_sheet/spooky
```
Comment (Jacquerel) [inline code/modules/clothing/suits/ghostsheet.dm:25]: now how will we prank mappers by having them accidentally place an admin item on their map...

Comment (carlarctg) [reply code/modules/clothing/suits/ghostsheet.dm:25]: look it was this or one guy murderboning everyone with incorporeality that doesn't even go away when dropped

File: code/modules/research/experimentor.dm (hunk 6 above)
Comment (carlarctg) [reply code/modules/research/experimentor.dm:881]: i thoguht itd be mor readablae :()

File: code/modules/research/experimentor.dm (hunk 5 above)
Comment (carlarctg) [reply code/modules/research/experimentor.dm:849]: yeah i know

File: code/modules/research/experimentor.dm (hunk 4 above)
Comment (carlarctg) [reply code/modules/research/experimentor.dm:814]: using uninstanced themes made it always turn things into iron. probably due to incorrect code

Comment (MrMelbert) [reply code/modules/research/experimentor.dm:814]: Idk I see
`terrain_theme = SSmaterials.dimensional_themes[pick(subtypesof(/datum/dimension_theme))]`
so I don't think you should be `new`ing here

File: code/__DEFINES/mobs.dm (hunk 8 above)
Comment (carlarctg) [reply code/__DEFINES/mobs.dm:387]: because actually fixing it is out of scope. this is just to make it not misleading

Comment (MrMelbert) [reply code/__DEFINES/mobs.dm:387]: removing it is equally out of scope as fixing it

**Review by ArcaneMusic — APPROVED** [review]

**Pull request merged**